tests/compile-fail/while_loop.rs

```diff
@@ -53,6 +53,33 @@ fn main() {
     while let Some(x) = y { // no error, obviously
         println!("{}", x);
     }
+
+    let mut iter = 1..20;
+    while let Option::Some(x) = iter.next() { //~ERROR this loop could be written as a `for` loop
+        println!("{}", x);
+    }
+
+    let mut iter = 1..20;
+    while let Some(x) = iter.next() { //~ERROR this loop could be written as a `for` loop
+        println!("{}", x);
+    }
```

Comment:
Whew! You had me worried I was mis-remembering how iterators worked – and I already had three cups of coffee, so my brain should sort of work currently... :smile:

Anyway, yeah, it shouldn't be too hard to detect and omit the lint in those situations.